### ISSUE TITLE
MySQL - one model produces records while another does not.

### ISSUE THREAD

**Issue by blthayer** [gridlab-d/gridlab-d#1198]
@ftuffner @eberleim 

Initial Checklist prior to submitting the issue. Check the bellow boxes to the best of your knowledge:

- [x] I have not found an open issue describing the same problem.
- [x] I went through "CONTRIBUTING.MD" before deciding to submit the issue.
- [x] While building GridLAB-D, I followed all steps as detailed in GridLAB-D Wiki.

------------------------------------------------------------------------------------------------------------

## Please describe what was the cause of the issue?
Given two similar models, one emits records to a MySQL database, while another does not.

GridLAB-D version: `GridLAB-D 4.1.0-18165 (Lugo [76964ea7:develop:Mod]) 64-bit LINUX RELEASE` (compiled with MySQL).

Please note this issue is a bit of an emergency for me, as I have demonstrations of my GridAPPS-D application coming up this week and in early November.

## Please provide the steps to reproduce.

I've attached two models in [models.zip](https://github.com/gridlab-d/gridlab-d/files/3725157/models.zip), `model_8500.glm` and `model_9500.glm`. 

### Summary
Running `model_8500.glm` creates records in MySQL, while running `model_9500.glm` does not.

Both of these models have two MySQL recorders, and a corresponding tape recorder and tape group_recorder, in order to record the same data in different places to help diagnose the issue.

### Modifications to the models
In order to get these models to run in your environment, you'll need to modify the `database` objects to match your MySQL server. This should be the **only** modification necessary for the models to run successfully.

### Running `model_8500.glm`
This model should create two tables: `triplex_8500` and `substation_8500`. Additionally, two files are created: `triplex_8500.csv` and `substation_8500.csv`.  

Here's the output I get when running the model:
```
root@eb31ba9998a1:/tmp# gridlabd model_8500.glm 
WARNING  [INIT] : object name '221-242079' does not follow strict naming rules and may not link correctly during load time
WARNING  [INIT] : last warning message was repeated 39 times
WARNING  [INIT] : Node:cap_capbank2a does not have the same nominal voltage as its parent - copying voltage from parent.
WARNING  [INIT] : last warning message was repeated 9 times
WARNING  [INIT] : Using a 2x2 z-matrix, instead of geometric values, is an under-determined system. Ground and/or neutral currents will be incorrect.
WARNING  [INIT] : last warning message was repeated 1176 times
WARNING  [INIT] : event_gen:external_event_handler does not have a metrics object as a parent -- metrics are not calculated.
WARNING  [INIT] : Node:sourcebus_meter is parented to a swing node and will get folded into it.
WARNING  [2019-10-14 14:30:50 UTC] : Regulator reg_FEEDER_REG has phase A at the maximum tap value
WARNING  [2019-10-14 14:30:50 UTC] : last warning message was repeated 11 times
WARNING  [2019-10-14 14:30:50 UTC] : Restoration object not detected!
WARNING  [2019-10-14 14:30:50 UTC] : Regulator reg_FEEDER_REG has phase A at the maximum tap value
WARNING  [2019-10-14 14:31:50 UTC] : last warning message was repeated 155 times
```

After running the model, you should see the data in the MySQL database:
```
mysql> select * from triplex_8500 limit 10;
+----+---------------------+----------------+-------------------------+
| id | t                   | name           | measured_voltage_12_mag |
+----+---------------------+----------------+-------------------------+
|  1 | 2019-10-14 14:30:50 | ld_251859c0    |              254.635642 |
|  2 | 2019-10-14 14:30:50 | ld_356790c0    |              254.179045 |
|  3 | 2019-10-14 14:30:50 | ld_226193698c0 |              309.677146 |
|  4 | 2019-10-14 14:30:50 | ld_21426205b0  |               276.85154 |
|  5 | 2019-10-14 14:30:50 | ld_351020c0    |              254.490023 |
|  6 | 2019-10-14 14:30:50 | ld_294788a0    |               264.64379 |
|  7 | 2019-10-14 14:30:50 | ld_21393454b0  |              262.896774 |
|  8 | 2019-10-14 14:30:50 | ld_356811a0    |              235.291472 |
|  9 | 2019-10-14 14:30:50 | ld_302392a0    |              258.475285 |
| 10 | 2019-10-14 14:30:50 | ld_337614b0    |              239.844168 |
+----+---------------------+----------------+-------------------------+
10 rows in set (0.00 sec)

mysql> select * from substation_8500 limit 10;
+----+---------------------+----------------------+---------------------+-------------------------+
| id | t                   | measured_real_energy | measured_real_power | measured_reactive_power |
+----+---------------------+----------------------+---------------------+-------------------------+
|  1 | 2019-10-14 14:30:50 |                    0 |                   0 |                       0 |
|  2 | 2019-10-14 14:30:55 |                    0 |                   0 |                       0 |
|  3 | 2019-10-14 14:31:00 |                    0 |                   0 |                       0 |
|  4 | 2019-10-14 14:31:05 |                    0 |                   0 |                       0 |
|  5 | 2019-10-14 14:31:10 |                    0 |                   0 |                       0 |
|  6 | 2019-10-14 14:31:15 |                    0 |                   0 |                       0 |
|  7 | 2019-10-14 14:31:20 |                    0 |                   0 |                       0 |
|  8 | 2019-10-14 14:31:25 |                    0 |                   0 |                       0 |
|  9 | 2019-10-14 14:31:30 |                    0 |                   0 |                       0 |
| 10 | 2019-10-14 14:31:35 |                    0 |                   0 |                       0 |
+----+---------------------+----------------------+---------------------+-------------------------+
10 rows in set (0.00 sec)
```

Similarly, the files should have data in them (showing results only for `substation_8500.csv` as the `triplex_8500.csv` file has a **ton** of columns):
```
root@eb31ba9998a1:/tmp# head -n 15 substation_8500.csv 
# file...... substation_8500.csv
# date...... Mon Oct 14 15:20:18 2019
# user...... (null)
# host...... (null)
# target.... meter 11047
# trigger... (none)
# interval.. 5
# limit..... -1
# timestamp,measured_real_energy, measured_real_power, measured_reactive_power
2019-10-14 14:30:50 UTC,+0,+0,+0
2019-10-14 14:30:55 UTC,+0,+0,+0
2019-10-14 14:31:00 UTC,+0,+0,+0
2019-10-14 14:31:05 UTC,+0,+0,+0
2019-10-14 14:31:10 UTC,+0,+0,+0
2019-10-14 14:31:15 UTC,+0,+0,+0
```
### Running `model_9500.glm`
Here's the output I get from running the model:

```
root@eb31ba9998a1:/tmp# gridlabd model_9500.glm 
WARNING  [INIT] : object name '221-242079' does not follow strict naming rules and may not link correctly during load time
WARNING  [INIT] : last warning message was repeated 39 times
WARNING  [INIT] : Node:cap_capbank2a does not have the same nominal voltage as its parent - copying voltage from parent.
WARNING  [INIT] : last warning message was repeated 9 times
WARNING  [INIT] : Using a 2x2 z-matrix, instead of geometric values, is an under-determined system. Ground and/or neutral currents will be incorrect.
WARNING  [INIT] : last warning message was repeated 1274 times
WARNING  [INIT] : event_gen:external_event_handler does not have a metrics object as a parent -- metrics are not calculated.
WARNING  [INIT] : Node:sourcebus_meter is parented to a swing node and will get folded into it.
WARNING  [INIT] : solarpanel: no climate data found, using static data
WARNING  [2019-10-14 14:24:32 UTC] : last warning message was repeated 353 times
WARNING  [2019-10-14 14:24:32 UTC] : Regulator reg_feeder_reg1a has phase A at the maximum tap value
WARNING  [2019-10-14 14:24:32 UTC] : last warning message was repeated 17 times
WARNING  [2019-10-14 14:24:32 UTC] : Restoration object not detected!
WARNING  [2019-10-14 14:24:32 UTC] : Unsupported phase on a possibly meshed node
WARNING  [2019-10-14 14:24:32 UTC] : Regulator reg_feeder_reg1a has phase A at the maximum tap value
WARNING  [2019-10-14 14:25:32 UTC] : last warning message was repeated 449 times
```

In this case, the tables get created in MySQL, but no records are inserted:
```
mysql> select * from triplex_9500;
Empty set (0.00 sec)

mysql> select * from substation_9500;
Empty set (0.00 sec)
```

However, our output files do indeed have data:
```
root@eb31ba9998a1:/tmp# head -n 15 substation_9500.csv 
# file...... substation_9500.csv
# date...... Mon Oct 14 15:33:02 2019
# user...... (null)
# host...... (null)
# target.... meter 14203
# trigger... (none)
# interval.. 5
# limit..... -1
# timestamp,measured_real_energy, measured_real_power, measured_reactive_power
2019-10-14 14:24:32 UTC,+0,+0,+0
2019-10-14 14:24:37 UTC,+0,+0,+0
2019-10-14 14:24:42 UTC,+0,+0,+0
2019-10-14 14:24:47 UTC,+0,+0,+0
2019-10-14 14:24:52 UTC,+0,+0,+0
2019-10-14 14:24:57 UTC,+0,+0,+0
```

### Bonus issue that I need solved:
Why does the substation energy and power measure as 0 in both models? 

If you run the following model, you do indeed get non-zero energy and power values. This model is designed to be a very simple sub-set of `model_9500.glm`, including only the swing node (`substation` object), a line from the swing node, and a load at the end of the line. The meter and recorder are setup similarly to how they are in `model_9500.glm` and `model_8500.glm`.

```
// Test file to ensure we get the expected results when parenting a
// meter to a substation, and then subsequently parenting a recorder to
// the meter. So that tests can run this with both the tape and mysql
// modules, the recorder itself and tape/mysql modules will be left
// out.

// Note the load object at the end of the model has very simple load
// values to make it easy to assert the correct values are coming out.
// We would expect the measured_real_power to always be 30,000 and the
// measured_reactive_power to always be 3,000.

#define VSOURCE=66400

module powerflow {
    solver_method NR;
}

module tape;

clock {
	timezone PST+8PDT;
	starttime '2001-01-01 00:00:00';
	stoptime '2001-01-01 00:01:00';
}

object substation {
  name "sourcebus";
  bustype SWING;
  phases ABCN;
  nominal_voltage 66395.28;
  base_power 12MVA;
  power_convergence_value 100VA;
  positive_sequence_voltage ${VSOURCE};
}

object meter {
  phases ABCN;
  nominal_voltage 66395.28;
  name "sourcebus_meter";
  parent "sourcebus";
};

object overhead_line {
	name "line_hvmv_sub_hsb";
	from "sourcebus_meter";
	to "hvmv115_hsb1";
	phases ABC;
	length 3.2809;
	continuous_rating_A 400.00;
	continuous_rating_B 400.00;
	continuous_rating_C 400.00;
	emergency_rating_A 400.00;
	emergency_rating_B 400.00;
	emergency_rating_C 400.00;
	configuration "lcon_hvmv_sub_hsb_ABC";
};

object line_configuration {
	name "lcon_hvmv_sub_hsb_ABC";
	z11 0.00000+14.7983j;
	c11 0.0000;
	z12 0.00000+0.00000j;
	c12 0.0000;
	z13 0.00000+0.00000j;
	c13 0.0000;
	z21 0.00000+0.00000j;
	c21 0.0000;
	z22 0.00000+14.7983j;
	c22 0.0000;
	z23 0.00000+0.00000j;
	c23 0.0000;
	z31 0.00000+0.00000j;
	c31 0.0000;
	z32 0.00000+0.00000j;
	c32 0.0000;
	z33 0.00000+14.7983j;
	c33 0.0000;
};

object load {
	name "hvmv115_hsb1";
	phases ABCN;
	nominal_voltage 66395.28;
	constant_power_A 10000+1000j;
	constant_power_B 10000+1000j;
	constant_power_C 10000+1000j;
};


// Tests should add a recorder that is similar to the following:
object recorder {
	file substation.csv;
	name substation_recorder;
	parent "sourcebus_meter";
	property "measured_real_energy, measured_real_power, measured_reactive_power";
	interval 5;
	limit -1;
};
```

And if we look at the output `substation.csv`:

```
root@eb31ba9998a1:/tmp# cat substation.csv 
# file...... substation.csv
# date...... Mon Oct 14 15:41:04 2019
# user...... (null)
# host...... (null)
# target.... meter 1
# trigger... (none)
# interval.. 5
# limit..... -1
# timestamp,measured_real_energy, measured_real_power, measured_reactive_power
2001-01-01 00:00:00 PST,+0,+30000,+3000
2001-01-01 00:00:05 PST,+41.6667,+30000,+3000
2001-01-01 00:00:10 PST,+83.3333,+30000,+3000
2001-01-01 00:00:15 PST,+125,+30000,+3000
2001-01-01 00:00:20 PST,+166.667,+30000,+3000
2001-01-01 00:00:25 PST,+208.333,+30000,+3000
2001-01-01 00:00:30 PST,+250,+30000,+3000
2001-01-01 00:00:35 PST,+291.667,+30000,+3000
2001-01-01 00:00:40 PST,+333.333,+30000,+3000
2001-01-01 00:00:45 PST,+375,+30000,+3000
2001-01-01 00:00:50 PST,+416.667,+30000,+3000
2001-01-01 00:00:55 PST,+458.333,+30000,+3000
```
-------------------------------------------------------------------------------------------------------------

**Comment by ftuffner** [comment]
Behavior confirmed on Windows build:
GridLAB-D 4.1.0-18171 (Lugo [090ffef2:bugfix/1198]) 64-bit WINDOWS DEBUG
8500-node file produces results, but the 9500-node file produces two empty tables.

**Comment by eberleim** [comment]
Patch for non-commit issue is now on branch feature/1198
I'll look more into the issue with the zero-values, it may be a formatting issue. (the +30000 type values, if submitted with the '+' could be a problem for mysql).

**Comment by blthayer** [comment]
@eberleim - I've tested your branch (bugfix/1198), and can confirm that running "model_9500" now does in fact push data into MySQL (for both the triplex and substation).

**Comment by blthayer** [comment]
@ftuffner - any chance you're able to look into the zero values portion? This is related to [1197](https://github.com/gridlab-d/gridlab-d/issues/1197#issue-506615238) (at least for me) - I believe that I had initially set up my code to change the line to come "from" the meter (instead of the substation object) since I was getting these zero values. 

Now that the method of setting the line to come "from" the meter doesn't work (at least for the 8500 and 9500 models), I'm in a pickle haha.

**Comment by ftuffner** [comment]
@blthayer There's a couple things going on here.  First off, the models you sent originally are giving zero because the line is in the wrong place -- you are outputting the meter, but the line is attached to the substation, so therefore, there is no "flow" through the meter, and it is in fact, zero.  That said, if you attach the line to the meter, there is the "topology issue" addressed in #1197.  So you basically have two different problems going on here, but updating these GLMs and running the fix in #1197 (soon to be merged into /develop) looks like it works.

**Comment by ftuffner** [comment]
Fix merged into /develop as part of PR #1200.  Closing.
